airbyte-integrations/connectors/source-commcare/source_commcare/source.py

```diff
@@ -34,15 +35,22 @@ def url_base(self) -> str:
     last_form_date = None
     schemas = {}
     unwantedfields = re.compile(r"^(case_|update_|meta|create_|commcare_).*$")
-
+  
     @property
     def dateformat(self):
         return "%Y-%m-%dT%H:%M:%S.%f"
 
     def scrubUnwantedFields(self, form):
-        newform = {k: v for k, v in form.items() if not self.unwantedfields.match(k)}
-        return newform
-
+        new_dict = {}
+        for key, value in form.items():
+            if key in self.form_fields:
+             continue
+            if isinstance(value, dict):
+             new_dict[key] = self.scrubUnwantedFields(value)
+            else:
+                new_dict[key] = value
+        return new_dict
```

Comment:
Here you're keeping the values from the `form_fields` which should be deleted right?

Reply:
This is why we changed to this. So it can be empty or user can choose which fields he doesn't want.